retrieval/discovery/marathon/marathon_test.go

```diff
@@ -40,11 +40,14 @@ func TestMarathonSDHandleError(t *testing.T) {
 	ch, md := newTestDiscovery(func(url string) (*AppList, error) {
 		return nil, errTesting
 	})
+	stop := make(chan struct{})
+	defer close(stop)
+
 	go func() {
 		select {
 		case tg := <-ch:
 			t.Fatalf("Got group: %s", tg)
-		default:
+		case <-stop:
```

Comment:
The `updateServices` sends to the channel in the background. If we closed it on our side, we could cause a panic.

Anyway, this PR is superseeded by #1735 anyway.